docs/GCSDoneFileMarker-postaction.md

```diff
@@ -0,0 +1,70 @@
+# Google Cloud Storage Done File Marker
+
+Description
+-----------
+This post-action plugin marks the end of a pipeline run by creating and storing an empty DONE (or SUCCESS) file in the 
+given GCS bucket upon a pipeline *completion*, *success*, or *failure*.
+
+This post-action plugin helps orchestrate various jobs which execution of one job is dependent on another job run 
+status.
+
+Credentials
+-----------
+If the plugin is run on a Google Cloud Dataproc cluster, the service account key does not need to be
+provided and can be set to 'auto-detect'.
+Credentials will be automatically read from the cluster environment.
+
+If the plugin is not run on a Dataproc cluster, the path to a service account key must be provided.
+The service account key can be found on the Dashboard in the Cloud Platform Console.
+Make sure the account key has permission to access BigQuery and Google Cloud Storage.
+The service account key file needs to be available on every node in your cluster and
+must be readable by all users running the job.
+
+Properties
+----------
+**Run Condition:** When to run the action. Must be *completion*, *success*, or *failure*. Defaults to *completion*.
+If set to *completion*, the action will be executed and a marker file will get created regardless of whether the
+ pipeline run succeeded or failed.
+If set to *success*, the action will get executed and the marker file will get created only if the pipeline run
+ succeeded.
+If set to *failure*, the action will get executed and the marker file will get created only if the pipeline run
+ failed.
+
+**Project ID**: Google Cloud Project ID, which uniquely identifies a project.
+It can be found on the Dashboard in the Google Cloud Platform Console.
+
+**Path** Google Cloud Storage path to the marker file. This takes the format `gs://<bucket>/file-marker-name`. 
```

Comment:
could this be in the format of `gs://<bucket>/directory/file-marker-name`? Does code require any special handling for creating directories?

Reply:
Subpaths are created automatically. No need for special handling. Changed the doc.

Reply:
mention the behavior for bucket. If bucket is not present it gets created